packages/pwa-kit-extension-sdk/src/react/classes/ApplicationExtension.ts

```diff
@@ -174,28 +174,21 @@ export class ApplicationExtension<
      * @throws Error if getComponentMap() is not defined.
      * @throws Error if the deserialized component cannot be found in the component map.
      */
-    private deserialize(): DeserializedExtension | null {
-        if (isServerSide() || typeof this.getRoutesAsync === 'undefined') {
+    public deserialize(componentMap: ComponentMap) {
+        if (typeof this.getRoutesAsync === 'undefined') {
             return null
         }
 
-        if (!this.getComponentMap) {
-            throw new Error(
-                `getComponentMap() must be defined when getRoutesAsync() is defined in the ${this.getName()} extension`
-            )
+        let serializedRoutes
+        if (!isServerSide()) {
+            console.log('JINSUUUU !isServerSide')
+            serializedRoutes = window.__EXTENSIONS__[this.getName()].routes
+        } else {
+            serializedRoutes = this._cachedRoutes
         }
```

Comment:
one finding is that on the server side we actually do have to deserialize the routes or at least the route we are currently on.

See here.